internal/utils/verify/verify.go

```diff
@@ -0,0 +1,470 @@
+// lucas zanotelli dos santos 2022-11-18
+// this code was mostly copied from https://github.com/stellar/go/blob/master/services/horizon/internal/ingest/verify.go
+
+package verify
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/guregu/null"
+	"github.com/stellar/stellar-etl/internal/transform"
+
+	"github.com/stellar/go/historyarchive"
+	"github.com/stellar/go/ingest"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/xdr"
+)
+
+// stateVerifierExpectedIngestionVersion defines a version of ingestion system
+// required by state verifier. This is done to prevent situations where
+// ingestion has been updated with new features but state verifier does not
+// check them.
+// There is a test that checks it, to fix it: update the actual `verifyState`
+// method instead of just updating this value!
+// const stateVerifierExpectedIngestionVersion = 15
+
+// verifyState is called as a go routine from pipeline post hook every 64
+// ledgers. It checks if the state is correct. If another go routine is already
+// running it exits.
+func VerifyState(
+	ctx context.Context,
+	transformedOutput transform.TransformedOutput,
+	archive historyarchive.ArchiveInterface,
+	ledgerSequence uint32,
+	verifyBatchSize int) (bool, error) {
+
+	stateReader, _ := ingest.NewCheckpointChangeReader(ctx, archive, ledgerSequence)
+	verifier := NewStateVerifier(stateReader, nil)
+
+	var keys []xdr.LedgerKey
+	keys, err := verifier.GetLedgerKeys(verifyBatchSize)
+	if err != nil {
+		return false, errors.Wrap(err, "verifier.GetLedgerKeys")
+	}
+
+	if len(keys) == 0 {
+		return false, errors.Wrap(err, "no keys")
+	}
+
+	err = addAccountsToStateVerifier(ctx, verifier, transformedOutput.Accounts, transformedOutput.Signers)
+	if err != nil {
+		return false, errors.Wrap(err, "addAccountsToStateVerifier failed")
+	}
+
+	err = addOffersToStateVerifier(ctx, verifier, transformedOutput.Offers)
+	if err != nil {
+		return false, errors.Wrap(err, "addOffersToStateVerifier failed")
+	}
+
+	err = addTrustLinesToStateVerifier(ctx, verifier, transformedOutput.Trustlines)
+	if err != nil {
+		return false, errors.Wrap(err, "addTrustLinesToStateVerifier failed")
+	}
+
+	err = addClaimableBalanceToStateVerifier(ctx, verifier, transformedOutput.Claimable_balances)
+	if err != nil {
+		return false, errors.Wrap(err, "addClaimableBalanceToStateVerifier failed")
+	}
+
+	err = addLiquidityPoolsToStateVerifier(ctx, verifier, transformedOutput.Liquidity_pools)
+	if err != nil {
+		return false, errors.Wrap(err, "addLiquidityPoolsToStateVerifier failed")
+	}
+
+	return true, nil
```

Comment:
There should be `Verify()` call after `Write`'ing all ledger entries.

Reply:
@bartekn calling `Verify()` against `StateVerifier.currentEntries` makes sense when these entries come from the Captive Core? I'm struggling to implement this method as it is because the logic I used in this file is different than [the original one](https://github.com/stellar/go/blob/master/services/horizon/internal/ingest/verify.go).

If implementing this is really necessary, could you shed some light on how to?